logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/LogStreamReaderTest.java

```diff
@@ -232,27 +236,48 @@ public void shouldSeekToMiddleOfBatch() {
   public void shouldIterateMultipleTimes() {
     // given
     final int eventCount = 500;
-    writeEvents(eventCount);
+    final var entries = IntStream.range(0, eventCount).mapToObj(TestEntry::ofKey).toList();
+    writer.tryWrite(entries);
 
     // when
+    var lastPosition = -1L;
     reader.seekToFirstEvent();
-    readerRule.assertEvents(eventCount, EVENT_VALUE);
+    for (int i = 0; i < entries.size(); i++) {
+      final var loggedEvent = readerRule.nextEvent();
+      assertThat(loggedEvent.getPosition()).isGreaterThan(lastPosition);
+      assertThat(loggedEvent.getKey()).isEqualTo(i);
+      assertThatEntry(entries.get(i)).matchesLoggedEvent(loggedEvent);
+      lastPosition = loggedEvent.getPosition();
+    }
```

Comment:
:wrench: Unless I'm wrong, this is just a duplicated block multiple times, right? We can extract that to a method.